packages/block-library/src/button/edit.js

```diff
@@ -292,12 +292,23 @@ function ButtonEdit( props ) {
 						...spacingProps.style,
 						...shadowProps.style,
 					} }
-					onSplit={ ( value ) =>
-						createBlock( 'core/button', {
-							...attributes,
-							text: value,
-						} )
-					}
+					onSplit={ ( value, isOriginal ) => {
+						let newAttributes;
```

Comment:
I would be also curious to better understand the changes applied here.

Reply:
Okay, I'll try to do my best from what I understood from my testings 😄 

When you use `onSplit` and call the [splitValue function](https://github.com/WordPress/gutenberg/blob/e58e179f56a48405ff3829be54a83bf1c473b193/packages/block-editor/src/components/rich-text/split-value.js#L11-L65), there are some things to take into account:

* When you click enter it splits the blocks into two.
* Where you have the cursor delimits, what's gonna be part of the first block (content to the left), and part of the new block (content to the right).
* I believe it actually creates two blocks: The original one and a new one. And this onSplit function runs for both of them.

With the previous logic, we were passing the attributes even if it wasn't the original block and its value it's empty. This happens when you click enter at the end of the block, with the cursor in the last character. You only want to preserve some value and the attributes when you actually split the text.

Actually, I just realized that we are preserving the metadata in the new block, which I believe it is not correct. For example, if I have a button and rename it to "Custom button" with text "My button", I put the cursor in the middle and I click Enter, the new block has the "Custom button" name as well. It should be easy to fix just checking if it is the original block or not.  I can work on that.

Reply:
Would it work this way?
```js
newAttributes = {
    ...attributes,
    metadata: isOriginal ? attributes.metadata : undefined,
    text: value,
  }; 
```

Reply:
I guess 🙂 Much better!

Reply:
This seems unrelated to the main problem? Maybe split this out in a separate PR?